Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -237,18 +242,27 @@ public async Task<IEnumerable<SubmissionForProfileServiceModel>> GetForProfileBy
         int take = 0)
     {
         var problem = await this.problemsDataService.GetWithProblemGroupById(problemId);
-
-        await this.ValidateUserCanViewResults(problem!, isOfficial);
+        var user = this.userProviderService.GetCurrentUser();
 
         var participant =
             await this.participantsDataService.GetByContestByUserAndByIsOfficial(
                 problem!.ProblemGroup.ContestId,
-                this.userProviderService.GetCurrentUser().Id!,
+                user.Id!,
                 isOfficial);
 
-        var userSubmissions = this.submissionsData
-            .GetAllByProblemAndParticipant(problemId, participant!.Id)
-            .MapCollection<SubmissionResultsServiceModel>();
+        var validationResult = this.submissionResultsValidationService.GetValidationResult((user, problem, participant));
+        if (!validationResult.IsValid)
+        {
+            throw new BusinessServiceException(validationResult.Message);
+        }
+
+        var userSubmissions = user.IsAdminOrLecturer || participant == null
+            ? this.submissionsData
```

Comment:
Isn't GetForProfileBy meant to get submissions from different contests that the user has participated in?

Reply:
We are getting here all of the submissions per problem, if the user is admin or lecturer. If the user is not admin or lecturer, then we get only the submissions for the participant for this problem. If the user is not admin or lecturer and is not a participant, then an error would be thrown.